CMakeLists.txt

```diff
@@ -41,6 +41,12 @@ set(SYMBOLSUFFIX ""
     CACHE STRING "suffix to add to ARPACK, BLAS and LAPACK function names")
 option(INTERFACE64 "use the 64-bit integer interface (ILP64) for ARPACK, BLAS and LAPACK")
 
+# 'make install' to the correct location, and also define
+# paths for target_include_directories and target_link_libraries
+include(GNUInstallDirs)
+set(ARPACK_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}/arpack-ng${ITF64SUFFIX}")
+set(ARPACK_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/arpackng${LIBSUFFIX}${ITF64SUFFIX}")
```

Comment:
Set `set(ARPACK_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/arpack-ng${LIBSUFFIX}${ITF64SUFFIX}")` _with_ dash for consistency? @sylvestre: what do you think?

Reply:
I just copied the original configuration. Making this change should lead to incompatibilities, specially if it is also not done consistently with the autoconf scripts. I leave it to your judgement to decide.

Reply:
> I just copied the original configuration.

Yes, got it.

> I leave it to your judgement to decide.

Weird ?!... From CI on master https://github.com/opencollab/arpack-ng/actions/runs/2895871392/jobs/4606355121#step:6:289, cmake seems to install include and libs in an arpack-ng (_with_ dash) directory so we should have `set(ARPACK_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/arpack-ng${LIBSUFFIX}${ITF64SUFFIX}")`, where autotools produces this install https://github.com/opencollab/arpack-ng/actions/runs/2895871392/jobs/4606354821#step:6:1430

Anyway, I'll set `set(ARPACK_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/arpack-ng${LIBSUFFIX}${ITF64SUFFIX}")`, _with_ dash, for consistency if the install tests (`tstCMakeInstall.sh` and tstAutotoolsInstall.sh``) pass: this way at least what worked before still work. I'll push this modification in a commit on top of yours: you can retrieve it by pulling at your side, and then, you'll can push on top of me.

Reply:
No, PLEASE, if you do so, find_package(arpackng) will NOT find the CMake configuration files! We have discussed this already. `find_package(arpackng)` searchers for `/usr/lib/cmake/arpackng*` directories. If you change the installation to `/usr/lib/cmake/arpack-ng`, then CMake will not find the `arpackng-config.cmake` file

Reply:
As I mentioned in the other pull request, this is one of the critical fixes in this patch. If this is changed, then the user must supply CMake with the installation directory for Arpack, which is *not* how things are done when using CMake. Users should be able to simply use `find_package(arpackng)` and let the build system find the packages. This is critical for allowing `arpackng` being built in systems such as `vcpkg` which is what I am trying to do.

Reply:
Changing the installation directory to `arpackng` just for the CMake files does not lead to any inconsistency, because the former CMake installation was simply broken.

Reply:
It must also be remarked that the autoconf tools are not capable of building useful CMake files at this moment -- they will never be able to reproduce what CMake is doing with their targets. Moreover, they are never installed in any of the platforms where I have seen arpack distributed; only the pkg-config file is available (e.g Debian, Ubuntu), so i do not think there is a need to keep both of them in sync in this aspect.